plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -104,18 +108,51 @@ export function getReviewData(formData: Record<string, any>, steps: Step[]) {
   return reviewData;
 }
 
-export const MultistepJsonForm = ({
-  steps,
-  formData,
-  onChange,
-  onReset,
-  onFinish,
-  fields,
-  widgets,
-}: Props) => {
+export const MultistepJsonForm = (props: Props) => {
+  const { formData, onChange, onReset, onFinish, fields, widgets } = props;
   const [activeStep, setActiveStep] = useState(0);
   const [disableButtons, setDisableButtons] = useState(false);
   const errorApi = useApi(errorApiRef);
+  const featureFlagApi = useApi(featureFlagsApiRef);
+  const featureFlagKey = 'backstage:featureFlag';
+  const filterOutProperties = (schema: JsonObject): JsonObject => {
+    const removedPropertyKeys: Array<string> = [];
+    if (schema.properties) {
+      schema.properties = Object.fromEntries(
+        Object.entries(schema.properties).filter(([key, value]) => {
+          if (value[featureFlagKey]) {
+            if (featureFlagApi.isActive(value[featureFlagKey])) {
+              return true;
+            }
+            removedPropertyKeys.push(key);
+            return false;
+          }
+          return true;
+        }),
+      );
+
+      // remove the feature flag property key from required if they are not active
+      schema.required = Array.isArray(schema.required)
+        ? schema.required?.filter(
+            r => !removedPropertyKeys.includes(r as string),
+          )
+        : schema.required;
+    }
+    return schema;
+  };
+
+  const steps = props.steps.filter(s => {
+    const featureFlag = s.schema[featureFlagKey];
+    if (
+      typeof featureFlag === 'string' &&
+      !featureFlagApi.isActive(featureFlag)
+    ) {
+      return null;
+    }
+    // filter out properties accordingly to the feature flag settings;
+    s.schema = filterOutProperties(s.schema);
```

Comment:
Oh I just realized something. This actually destroys the input. That's kinda risky and will surprise the caller. You probably want to change this into a filter + map.

```ts
const steps = props.steps
  .filter(step => {
    const featureFlag = step.schema[featureFlagKey];
    return typeof featureFlag !== 'string' || featureFlagApi.isActive(featureFlag);
  })
  .map(filterOutProperties);
```

and then make sure that that function doesn't change anything, maybe start it off with 

```ts
import { clone } from 'lodash';
// ...
const output = clone(schema);
// now you can do whatever you wan to the output
return output;
```